### PR TITLE
Add dependency verifier and color constants

### PR DESCRIPTION
**Pull request by viktorprogger**
Added dependency verifier which tries to execute `composer update` for all the packages and then says what's ok and what is not. Command logfile `runtime/composer-verify.log` contains `composer update` log for failed packages.
_In *nix it will work in 5 forks and in windows it will work in one process._

**Needs to be tested on Windows (I haven't an ability to do this).**

Example:
image

### CODE REVIEW

**Comment by viktorprogger** [comment]
I think I should add an abiliity to verify dependencies for alone package provided with an argument.

**Comment by samdark** [comment]
In case of error, can we output error text right away?

**Comment by samdark** [comment]
Also, using unicode looks great: https://github.com/yiisoft/yii-dev/blob/master/commands/ReplicateCommand.php#L56

**Comment by viktorprogger** [comment]
This is done, but I'm getting strange behavior with forks:
1. 4 of 5 forks reports they are exited right after they are created
1. After `pcntl_wait` reports all forks are returned I'm still getting output from forks.

So, the whole functionality is working, but with strange behavior while forking processes.

**Comment by viktorprogger** [comment]
The bug is fixed :)

**Comment by viktorprogger** [comment]
Ready for review

File: commands/DependencyVerifyCommand.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        return $result;
+    }
+
+    private function fork(): bool
+    {
+        if (function_exists('pcntl_fork')) {
+            $pid = pcntl_fork();
+
+            if ($pid < 0) {
+                stdoutln("Cannot fork", Color::RED);
```
Comment (samdark) [inline commands/DependencyVerifyCommand.php]:
```suggestion
                stderrln("Cannot fork", Color::RED);
```

**Comment by samdark** [comment]
Code looks fine and it works but I have questions.

1. It looks like the new command is the same as existing `update` except output format.
2. Parallelization could be applied to any command dealing with many repositories at once such as `install`, `update` or `pull`.

**Comment by samdark** [comment]
@viktorprogger https://github.com/yiisoft/yii-dev/pull/9#issuecomment-508269436?

**Comment by viktorprogger** [comment]
1. It's helpful when you want to test if there is an installable set of packages
2. There are some bugs in my realization of parallelization (`flock` doesn't seem to be atomic). I I'll remove it from this PR and make one another if I'll find a way out of this issue.

**Comment by rob006** [comment]
`composer update` has `--dry-run` switch - wouldn't be easier to use it and parse output instead of doing some voodoo magic with locks and `pcntl_fork()`? :)

**Comment by viktorprogger** [comment]
1. Damn me, I've forgot about `dry-run`. Thnx, Rob.
1. We still can parallelize tasks, so it will require locks and forks.
1. This PR is out of date. I'll make other PRs instead of it (the first is #15).

**Comment by rob006** [comment]
There is also `composer outdated`.

>     2\. We still can parallelize tasks, so it will require locks and forks.

I'm not sure what do you want to parallelize, but running multiple `composer update` commands seems to be really bad idea. Update is really memory-intensive, having 5 concurrent updates may give memory-related errors on most of machines.

**Pull request closed without merging**